includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1048,6 +1048,27 @@ public function create_intent( $order, $prepared_source ) {
 			$request['customer'] = $prepared_source->customer;
 		}
 
+		/**
+		 * Filter the return value of the WC_Payment_Gateway_CC::generate_create_intent_request.
+		 *
+		 * @since 3.1.0
+		 * @param array $request
+		 * @param WC_Order $order
+		 * @param object $source
+		 */
+		return apply_filters( 'wc_stripe_generate_create_intent_request', $request, $order, $prepared_source );
```

Comment:
Should we add a test to make sure that the filtered value is used for the `payments_intent` request? To avoid regressions that could lead to subscriptions not being marked as `setup_future_usage: 'off_session'`.

Reply:
Thank you for the filter @RadoslavGeorgiev 🎉 

Added it here: https://github.com/Automattic/woocommerce.com/pull/6386/files#diff-b2bb4966d53e7a3f9e440111ffc09354R54

For reference, this Payments Intent API call resulted in in a successful checkout.
```
payment_intents request: Array
(
    [source] => card_1FALcAESYdNq2OFAjBBZeylx
    [amount] => 5586
    [currency] => usd
    [description] => WooCommerce - Order 5101996149
    [metadata] => Array
        (
            [customer_name] => Dwain Maralack
            [customer_email] => aviva+vendor@automattic.com
            [order_id] => 5101996149
        )

    [statement_descriptor] => WooComm 5101996149
    [capture_method] => automatic
    [payment_method_types] => Array
        (
            [0] => card
        )

    [customer] => cus_FfFC7L8qef5DXU
    [setup_future_usage] => off_session
)
```

Reply:
> Should we add a test to make sure that the filtered value is used for the payments_intent request? To avoid regressions that could lead to subscriptions not being marked as setup_future_usage: 'off_session'.

We don't need to. The filter is in the abstract/base class, but `setup_future_usage` is added afterwards in the `WC_Stripe_Subs_Compat` class.